Jenkinsfile

```diff
@@ -16,60 +14,65 @@ pipeline {
     timestamps()
     timeout(time: 30, unit: 'MINUTES')
   }
-  environment {
-    PYTEST_PROCESSES = "${PYTEST_PROCESSES ?: "auto"}"
-    PYTEST_ADDOPTS =
-      "-n=${PYTEST_PROCESSES} " +
-      "--tb=short " +
-      "--color=yes " +
-      "--driver=SauceLabs " +
-      "--variables=capabilities.json"
-    PULSE = credentials('PULSE')
-    SAUCELABS = credentials('SAUCELABS')
-  }
   stages {
     stage('Lint') {
+      agent {
+        dockerfile true
+      }
       steps {
         sh "flake8"
       }
     }
     stage('Test') {
+      agent {
+        dockerfile true
+      }
+      environment {
+        PYTEST_PROCESSES = "${PYTEST_PROCESSES ?: "auto"}"
+        PULSE = credentials('PULSE')
+        SAUCELABS = credentials('SAUCELABS')
+      }
       steps {
         writeCapabilities(capabilities, 'capabilities.json')
-        sh "pytest --junit-xml=results/py27.xml " +
-          "--html=results/py27.html --self-contained-html " +
-          "--log-raw=results/py27_raw.txt " +
-          "--log-tbpl=results/py27_tbpl.txt"
+        sh "pytest -n=${PYTEST_PROCESSES} --tb=short --color=yes --driver=SauceLabs --variables=capabilities.json " +
+        "--junit-xml=results/junit.xml " +
+        "--html=results/index.html --self-contained-html " +
```

Comment:
As these are related arguments, I think it's fine for them to be on the same line, though if you want to split them for consistency that wfm.